### PR TITLE
#161779880 users are able to report articles.

### PR DESCRIPTION
**Pull request by gbols**
#### What does this PR do?
it enables users to report articles.
#### Description of Task to be completed?

- It enables users to post report about articles.

- admin can query the reports of a specific article

- admin can query all the reports on the platform

#### How should this be manually tested?
git clone `ft-report-an-article-161779880 `
npm install
npm test
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#[161779880](https://www.pivotaltracker.com/story/show/161779880)
#### Screenshots (if appropriate)
screen shot 2018-12-20 at 3 43 50 pm

### CODE REVIEW

**Review by davidshare — APPROVED** [review]
LGTM

**Review by VictoriaOffoma — CHANGES_REQUESTED** [review]
findArticle sounds like a name of a function. Use something that shows the value contained in the variable e.g "article, reported article"

File: server/controllers/reportArticleController.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            if(!(isFieldValid(report, 15))){
+                return res.status(400).send({
+                    success:false,
+                    message:'report cant be less than 15 characters'
+                });
+            }
+            const findArticle = await Article.findOne({
+                where: { id: articleId}
+            });
+            if (!findArticle) {
```
Comment (VictoriaOffoma) [inline server/controllers/reportArticleController.js]: Change the variable "findArticle" to something more descriptive like "article" or "reportedArticle".

File: server/helpers/validationHelper.js [hunk 2]
```diff
@@ -51,11 +51,24 @@ const  isSocialMediaEmail = (email) => {
   return email.includes('@facebook.com') || email.includes('@twitter.com');
 };
 
+/**
+ *
+ * @param {string} input argument to be checked
+ * @param {integer} length of the string to make it valid
+ * @returns {boolean} a values of true or false is returned
+ */
+const isFieldValid = (input, length = 1) => {
```
Comment (VictoriaOffoma) [inline server/helpers/validationHelper.js:72]: The "isFieldValid" is not descriptive as it does not tell us what field is or what field refers to.

**Review by VictoriaOffoma — APPROVED** [review]

**Pull request merged**